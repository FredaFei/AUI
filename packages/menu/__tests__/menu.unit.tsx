import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import Menu from '../menu'
import MenuItem from '../menuItem'
import SubMenu from '../subMenu'

describe('测试 Menu 以下行为', () => {
  it('Menu 存在', () => {
    const json = renderer.create(<Menu>
      <MenuItem name="menu1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Menu 参数 layout 默认值为“horizontal”', () => {
    const wrapper = mount(<Menu>
      <MenuItem name="menu1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    expect(wrapper.props().layout).toEqual('horizontal')
  })
  it('Menu 设置 selected', () => {
    const wrapper = mount(<Menu selected="menu3-2-2">
      <SubMenu title="sub1" name="sub1">
        <MenuItem name="menu3-1">menu 3--1</MenuItem>
        <MenuItem name="menu3-2">menu 3--2</MenuItem>
        <MenuItem name="menu3-3">menu 3--3</MenuItem>
        <SubMenu title="submenu 2" name="sub2">
          <MenuItem name="menu3-2-1">menu 3-2-1</MenuItem>
          <MenuItem name="menu3-2-2">menu 3-2-2</MenuItem>
          <MenuItem name="menu3-3-3">menu 3-2-3</MenuItem>
        </SubMenu>
      </SubMenu>
      <MenuItem name="menu1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    expect(wrapper.props().selected).toEqual('menu3-2-2')
  })
  it('Menu 点击menuItem 触发onChange', () => {
    const fn = jest.fn()
    const wrapper = mount(<Menu selected="menu3-2-2" onChange={fn}>
      <SubMenu title="sub1" name="sub1">
        <MenuItem name="menu3-1">menu 3--1</MenuItem>
        <MenuItem name="menu3-2">menu 3--2</MenuItem>
        <MenuItem name="menu3-3">menu 3--3</MenuItem>
        <SubMenu title="submenu 2" name="sub2">
          <MenuItem name="menu3-2-1">menu 3-2-1</MenuItem>
          <MenuItem name="menu3-2-2">menu 3-2-2</MenuItem>
          <MenuItem name="menu3-3-3">menu 3-2-3</MenuItem>
        </SubMenu>
      </SubMenu>
      <MenuItem name="menu1" className="test-menu-1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    wrapper.find('.test-menu-1').at(0).simulate('click')
    expect(fn).toHaveBeenCalledWith('menu1')
  })

  it('Menu 点击menuItem 触发onChange', () => {
    const fn = jest.fn()
    const wrapper = mount(<Menu selected="menu3-2-2" onChange={fn}>
      <SubMenu title="sub1" name="sub1">
        <MenuItem name="menu3-1">menu 3--1</MenuItem>
        <MenuItem name="menu3-2">menu 3--2</MenuItem>
        <MenuItem name="menu3-3">menu 3--3</MenuItem>
        <SubMenu title="submenu 2" name="sub2">
          <MenuItem name="menu3-2-1">menu 3-2-1</MenuItem>
          <MenuItem name="menu3-2-2">menu 3-2-2</MenuItem>
          <MenuItem name="menu3-3-3">menu 3-2-3</MenuItem>
        </SubMenu>
      </SubMenu>
      <MenuItem name="menu1" className="test-menu-1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    wrapper.find('.test-menu-1').at(0).simulate('click')
    expect(fn).toHaveBeenCalledWith('menu1')
  })

  it('MenuItem 设置 disalbed 值为“true”', () => {
    const fn = jest.fn()
    const wrapper = mount(<Menu onChange={fn}>
      <MenuItem name="menu1" disabled>menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    wrapper.find('.am-menuItem').at(0).simulate('click')
    expect(fn).not.toHaveBeenCalled()
    wrapper.find('.am-menuItem').at(1).simulate('click')
    expect(fn).toHaveBeenCalledWith('menu2')
  })
  it('SubMenu 设置 disalbed 值为“true”', () => {
    const fn = jest.fn()
    const wrapper = mount(<Menu onChange={fn}>
      <SubMenu title="sub1" name="sub1" disabled>
        <MenuItem name="menu3-1">menu 3--1</MenuItem>
        <MenuItem name="menu3-2">menu 3--2</MenuItem>
        <MenuItem name="menu3-3">menu 3--3</MenuItem>
        <SubMenu title="submenu 2" name="sub2">
          <MenuItem name="menu3-2-1">menu 3-2-1</MenuItem>
          <MenuItem name="menu3-2-2">menu 3-2-2</MenuItem>
          <MenuItem name="menu3-3-3">menu 3-2-3</MenuItem>
        </SubMenu>
      </SubMenu>
      <MenuItem name="menu1" disabled>menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    wrapper.find('.am-subMenu-label').at(0).simulate('click')
    expect(fn).not.toHaveBeenCalled()
  })

  it('Menu 设置 自定义类名', () => {
    const wrapper = shallow(<Menu className="test">
      <MenuItem name="menu1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    expect(wrapper.find('.am-menu-wrapper').hasClass('test')).toEqual(true)
  })
  it('Menu 设置 自定义样式', () => {
    const wrapper = shallow(<Menu className="test" style={{color: 'red'}}>
      <MenuItem name="menu1">menu 1</MenuItem>
      <MenuItem name="menu2">menu 2</MenuItem>
    </Menu>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})