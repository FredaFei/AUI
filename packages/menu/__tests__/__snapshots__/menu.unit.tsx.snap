// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Menu 以下行为 Menu 存在 1`] = `
<div
  className="am-menu-wrapper horizontal "
>
  <div
    className="am-menuItem "
    onClick={[Function]}
  >
    menu 1
  </div>
  <div
    className="am-menuItem "
    onClick={[Function]}
  >
    menu 2
  </div>
</div>
`;

exports[`测试 Menu 以下行为 Menu 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-menu-wrapper horizontal test",
      "style": "color:red",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-menuItem ",
        },
        "children": Array [
          Object {
            "data": "menu 1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "am-menuItem ",
          },
          "children": Array [
            Object {
              "data": "menu 2",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "am-menuItem ",
        },
        "children": Array [
          Object {
            "data": "menu 2",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "am-menuItem ",
          },
          "children": Array [
            Object {
              "data": "menu 1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
